src/aggregation/metric/top_hits.rs

```diff
@@ -0,0 +1,822 @@
+                crate::TantivyError::SchemaError(format!(
+                    "Invalid regex '{}' in docvalue_fields: {}",
+                    glob, e
+                ))
+            })
+        };
+        let resolved_fields = self
+            .doc_value_fields
+            .into_iter()
+            .map(|field| {
+                if !field.contains('*') && reader.iter_columns()?.any(|(name, _)| name == field) {
+                    return Ok(vec![field]);
+                }
+
+                let pattern = globbed_string_to_regex(&field)?;
+                Ok(reader
+                    .iter_columns()?
+                    .filter(|(name, _)| pattern.is_match(name))
+                    .map(|(name, _)| name)
+                    .collect::<Vec<_>>())
+            })
+            .collect::<crate::Result<Vec<_>>>()?
+            .into_iter()
+            .flatten()
+            .collect();
+
+        Ok(Self {
+            doc_value_fields: resolved_fields,
+        })
+    }
+
+    fn get_document_field_data(
+        &self,
+        accessors: &HashMap<String, DynamicColumn>,
+        doc_id: DocId,
+    ) -> FieldRetrivalResult {
+        let dvf = self
+            .doc_value_fields
+            .iter()
+            .map(|field| {
+                let accessor = accessors
+                    .get(field)
+                    .expect("could not find field in accessors");
+
+                let value: OwnedValue = match accessor {
+                    DynamicColumn::U64(accessor) => accessor
+                        .values_for_doc(doc_id)
+                        .next()
+                        .map_or(OwnedValue::Null, OwnedValue::U64),
+                    DynamicColumn::I64(accessor) => accessor
+                        .values_for_doc(doc_id)
+                        .next()
+                        .map_or(OwnedValue::Null, OwnedValue::I64),
+                    DynamicColumn::F64(accessor) => accessor
+                        .values_for_doc(doc_id)
+                        .next()
+                        .map_or(OwnedValue::Null, OwnedValue::F64),
+                    DynamicColumn::Bytes(accessor) => 'l: {
+                        let Some(term_ord) = accessor.term_ords(doc_id).next() else {
+                            break 'l OwnedValue::Null;
+                        };
+                        let mut buffer = vec![];
+                        match accessor
+                            .ord_to_bytes(term_ord, &mut buffer)
+                            .expect("could not find term_ord in the term dictionary")
+                        {
+                            false => OwnedValue::Null,
+                            true => OwnedValue::Bytes(buffer),
+                        }
+                    }
+                    DynamicColumn::Str(accessor) => 'l: {
+                        let Some(term_ord) = accessor.term_ords(doc_id).next() else {
+                            break 'l OwnedValue::Null;
```

Comment:
This is hard to read

Reply:
Removed with [d760c6c](https://github.com/quickwit-oss/tantivy/pull/2198/commits/d760c6c840b050f1ca3053af5b6f68c2f4afadcb)